lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -2163,6 +2163,66 @@ static bool foldMuxOfUniformArrays(MuxOp op, PatternRewriter &rewriter) {
   return true;
 }
 
+// Fold trees of muxes selecting an element from 2^N leafs into an array get.
+static bool foldMuxTree(MuxOp op, PatternRewriter &rewriter) {
+  SmallVector<Value> conditions;
+
+  // Walk down a single branch of the tree to find the deepest sequence
+  // of conditions to consider. This will be narrowed down later.
+  for (auto mux = op; mux; mux = mux.getFalseValue().getDefiningOp<MuxOp>())
+    conditions.push_back(mux.getCond());
```

Comment:
I believe this pattern as-is changes x behavior.  arra[x] = x, while mux(x,t,f) = merge(t,f).